### PR TITLE
nmstate: new function for default interface setting

### PR DESCRIPTION
**Pull request by ajaggapa**
- Add `WithEthernetDualStackInterface` method to PolicyBuilder that configures an ethernet interface with state "up", IPv4 enabled, and IPv6 enabled on a NodeNetworkConfigurationPolicy
- Add unit tests covering valid interface name, empty interface name, and invalid builder scenarios

## Summary by CodeRabbit

* **New Features**
  * Added support for configuring Ethernet interfaces with both IPv4 and IPv6 enabled.
  * Added validation for interface names and policy selectors when creating dual-stack interface configurations.

### CODE REVIEW

File: pkg/nmstate/policy_test.go [hunk 1]
```diff
@@ -870,6 +870,54 @@ func TestPolicyWithEthernetIPv6LinkLocalInterface(t *testing.T) {
...
+						Ipv4: InterfaceIpv4{
+							Enabled: true,
+						},
+						Ipv6: InterfaceIpv6{
+							Enabled: true,
+						},
+					},
+				},
+			}, desireState)
+		}
```
[the remark this answers is not in the document]
Comment (ajaggapa) [reply pkg/nmstate/policy_test.go:917]: Limiting the unit test to this function usage. So it ignoring this.

**Review by cdvultur — APPROVED** [review]
lgtm

**Review by klaskosk — APPROVED** [review]
